dotnet/src/Extism.Sdk/Plugin.cs

```diff
@@ -6,51 +6,43 @@ namespace Extism.Sdk.Native;
 /// <summary>
 /// Represents a WASM Extism plugin.
 /// </summary>
-public class Plugin : IDisposable
+public unsafe class Plugin : IDisposable
 {
     private const int DisposedMarker = 1;
 
-    private readonly Context _context;
     private readonly HostFunction[] _functions;
     private int _disposed;
 
+    /// <summary>
+    /// Native pointer to the Extism Plugin.
+    /// </summary>
+    internal LibExtism.ExtismPlugin* NativeHandle { get; }
+
     /// <summary>
     /// Create a and load a plug-in
-    /// Using this constructor will give the plug-in it's own internal Context
     /// </summary>
     /// <param name="wasm">A WASM module (wat or wasm) or a JSON encoded manifest.</param>
     /// <param name="functions">List of host functions expected by the plugin.</param>
     /// <param name="withWasi">Enable/Disable WASI.</param>
-    public static Plugin Create(ReadOnlySpan<byte> wasm, HostFunction[] functions, bool withWasi) {
-        var context = new Context();
-        return context.CreatePlugin(wasm, functions, withWasi);
-    }
-
-    internal Plugin(Context context, HostFunction[] functions, int index)
-    {
-        _context = context;
+    public Plugin(ReadOnlySpan<byte> wasm, HostFunction[] functions, bool withWasi) {
         _functions = functions;
-        Index = index;
-    }
+        var functionHandles = functions.Select(f => f.NativeHandle).ToArray();
 
-    /// <summary>
-    /// A pointer to the native Plugin struct.
-    /// </summary>
-    internal int Index { get; }
-
-    /// <summary>
-    /// Update a plugin, keeping the existing ID.
-    /// </summary>
-    /// <param name="wasm">The plugin WASM bytes.</param>
-    /// <param name="withWasi">Enable/Disable WASI.</param>
-    unsafe public bool Update(ReadOnlySpan<byte> wasm, bool withWasi)
-    {
-        CheckNotDisposed();
-
-        var functions = _functions.Select(f => f.NativeHandle).ToArray();
-        fixed (byte* wasmPtr = wasm)
+        unsafe
         {
-            return LibExtism.extism_plugin_update(_context.NativeHandle, Index, wasmPtr, wasm.Length, functions, 0, withWasi);
+            fixed (byte* wasmPtr = wasm)
+            fixed (IntPtr* functionsPtr = functionHandles)
+            {
+                NativeHandle = LibExtism.extism_plugin_new(wasmPtr, wasm.Length, functionsPtr, functions.Length, withWasi, null);
```

Comment:
@mhmd-azeez - the `extism_plugin_new` function now takes a final parameter which is `char **errormsg`,the way other SDKs have been doing this is to pass a reference to `char*` - I'm not sure how this should be done in C#, do you have any ideas?

Reply:
Same thing works in dotnet too:

```
_functions = functions;
var functionHandles = functions.Select(f => f.NativeHandle).ToArray();

unsafe
{
    fixed (byte* wasmPtr = wasm)
    fixed (IntPtr* functionsPtr = functionHandles)
    {
        char** errmsg = null;
        NativeHandle = LibExtism.extism_plugin_new(wasmPtr, wasm.Length, functionsPtr, functions.Length, withWasi, errmsg);
        if (NativeHandle == null)
        {
            string error = "Unable to create plugin";
            if (errmsg != null)
            {
                IntPtr err = (IntPtr)(*errmsg);
                error = Marshal.PtrToStringUTF8(err);
                LibExtism.extism_plugin_new_error_free(*errmsg);
            }

            throw new ExtismException(error);
        }
    }
}

```

Pushed a commit has the changes

Reply:
Oh great! Thank you